Utils/PluginActivator/source/COMRPCStarter.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "COMRPCStarter.h"
+
+#include "Log.h"
+
+#include <chrono>
+#include <thread>
+
+COMRPCStarter::COMRPCStarter(const string& pluginName)
+    : mPluginName(pluginName)
+    , mEngine(Core::ProxyType<RPC::InvokeServerType<1, 0, 4>>::Create())
+    , mClient(Core::ProxyType<RPC::CommunicatorClient>::Create(getConnectionEndpoint(),
+          Core::ProxyType<Core::IIPCServer>(mEngine)))
+{
+    // Announce our arrival
+    mEngine->Announcements(mClient->Announcement());
+
+    if (!mClient.IsValid()) {
+        LOG_ERROR(pluginName.c_str(), "Failed to create valid COM-RPC client");
+    }
+}
+
+COMRPCStarter::~COMRPCStarter()
+{
+    // Disconnect from Thunder and clean up
+    mClient->Close(RPC::CommunicationTimeOut);
+    if (mClient.IsValid()) {
+        mClient.Release();
+    }
+
+    Core::Singleton::Dispose();
+}
+
+/**
+ * @brief Attempt to activate the plugin and automatically retry on failure
+ *
+ * @param[in]   maxRetries      Maximum amount of times to retry activation if it fails
+ * @param[in]   retryDelayMs    Delay in ms between retry attempts
+ *
+ * @return True if plugin successfully activated, false if failed to activate
+ */
+bool COMRPCStarter::activatePlugin(const int maxRetries, const int retryDelayMs)
+{
+    // Attempt to open the plugin shell
+    bool success = false;
+    int currentRetry = 1;
+
+    PluginHost::IShell* shell = nullptr;
+
+    while (!success && currentRetry <= maxRetries) {
+        LOG_INF(mPluginName.c_str(), "Attempting to activate plugin - attempt %d/%d", currentRetry, maxRetries);
+
+        auto start = std::chrono::steady_clock::now();
```

Comment:
Minor, just a remark to know why there is a Core::Time.. As Thunder (e.g. WPEFramework, w.g. WebBridge) started out on a platform which required to work with compiler GCC 4.3 on MIPS, not supporting all STD C++ 11 features, the access to std::chrono was not available hence why there is an abstraction, called Core::Time. Although we are now on C++11 (we left GCC4.3 behind us since WebKit enforced now GCC6 or higher :-)  it is a perfect solution to use chrono but it is a mixture of time functionalities.
Question is a bit what to do with the new standardised functionality of C++11...Lets discuss next week..

Reply:
Switched to `Core::Time` for consistency with rest of codebase for now